v4/weaviate/weaviate_client.go

```diff
@@ -33,6 +34,19 @@ type Config struct {
 	Headers map[string]string
 }
 
+func NewConfig(host string, scheme string, authConfig auth.Config, headers map[string]string) (*Config, error) {
+	var client *http.Client = nil
```

Comment:
if you declare a pointer type, it will already hold a value of `nil` by default, no need to manually assign:
```
var client *http.Client
```

Reply:
changed